src/service/presence/mod.rs

```diff
@@ -170,6 +170,44 @@ impl Service {
 		self.db.remove_presence(user_id).await;
 	}
 
+	// Unset online/unavailable presence to offline on startup
+	pub async fn unset_all_presence(&self) -> Result<()> {
+		for user_id in &self
+			.services
+			.users
+			.list_local_users()
+			.map(UserId::to_owned)
```

Comment:
I think this can be optimised to avoid needing to call `to_owned`, but might be wrong. I'll try messing with it and see myself.

Reply:
> I think this can be optimised to avoid needing to call `to_owned`, but might be wrong. I'll try messing with it and see myself.

I think reference should do the job. I just borrowed this logic from another part of the codebase.

Reply:
Btw, I just remembered that we should also unset statuses for all users, including those federated. Maybe I could call db for fetching users directly?

Reply:
The startup performance impact will be very bad if we include federated users without further significant database optimisations, making this use pure references only instead of copying/to_owned, and optimising this code more.

I think it's best that is left out of this PR for now.

Reply:
I think I can optimise enough to allow clearing remote presences on startup for remote users without startup performance taking a toll.

Reply:
I just realized that the best solution may just be to store presence state in memory, and not persisting it to the database, since they will be removed on startup. Will federated presence information taking up too much memory?

Reply:
Yes it will. Presence is designed to track tens of thousands of remotes users. Presence contents can be large such as status messages. Storing them in memory is definitely not a valid option.

Reply:
> I think this can be optimised to avoid needing to call `to_owned`, but might be wrong. I'll try messing with it and see myself

you have to call `to_owned()` here before `collect()`